python/ray/serve/tests/test_cli.py

```diff
@@ -33,7 +33,38 @@ def assert_deployments_live(ids: List[DeploymentID]):
 
 def test_start_shutdown(ray_start_stop):
     subprocess.check_output(["serve", "start"])
-    subprocess.check_output(["serve", "shutdown", "-y"])
+    # deploy a simple app
+    import_path = "ray.serve.tests.test_config_files.arg_builders.build_echo_app"
+    subprocess.check_output(["serve", "deploy", import_path])
+    wait_for_condition(
+        check_http_response,
+        expected_text="DEFAULT",
+        timeout=15,
+    )
```

Comment:
There's actually a `click.testing`  module built for this purpose that we can use rather than subprocess: https://click.palletsprojects.com/en/stable/testing/